packages/stable/src/api/dataPoints/dataPointsApi.ts

```diff
@@ -121,4 +174,36 @@ export class DataPointsAPI extends BaseResourceAPI<
   }
 }
 
+interface MonthInfo {
+  start_date: Timestamp;
+  end_date: Timestamp;
+  number_of_days: number;
+}
+
+const getMonthsBetweenDates = (startDate: Timestamp | string, endDate: Timestamp | string): MonthInfo[] => {
+  const result: MonthInfo[] = [];
+
+  let currentMonth = new Date(startDate);
+  endDate = new Date(endDate);
+
+  while (currentMonth <= endDate) {
+    const year = currentMonth.getFullYear();
+    const month = currentMonth.getMonth() + 1; // Months are zero-indexed
+
+    const firstDay = new Date(year, month - 1, 1);
+    const lastDay = new Date(year, month, 0);
+
+    result.push({
+      start_date: firstDay.getTime(),
+      end_date: lastDay.getTime(),
+      number_of_days: lastDay.getDate() - firstDay.getDate() + 1,
+    });
+
+    // Move to the next month
+    currentMonth.setMonth(currentMonth.getMonth() + 1);
```

Comment:
This works between years? e.g. Dec 2022 will give you Jan 2023?

Reply:
It seems working, I'll write a test case for it

Reply:
works as expected, tested [here](https://github.com/cognitedata/cognite-sdk-js/pull/1034/files#diff-04fc6dab9e7c7cdac19f22741d90516fa63d141ad8a1a57d5e6a18ae3cd976bcR181)